### PR TITLE
Wip/stage workflow

### PR DESCRIPTION
**Pull request by hannalee2**
Implemented:

- Reticle Detection
- Probe Detection
- Update .toml and setup.py file to upload PyPI

Tested on Rig2 in the lab. The precision is not high enough for production yet, but updates will continue in the future.

### CODE REVIEW

File: parallax/probe_calibration.py [hunk 1]
```diff
@@ -3,125 +3,296 @@
...
         self.stages = {}
-        self.local_points = []
-        self.global_points = []
+        self.df = None
         self.inliers = []
-        self.transform_matrix = None
-        self.error_min = 1000
-        
+        self.stage = None
+        self.threshold_min_mix = 2000 #TODO
```
Comment (jsiegle) [inline parallax/probe_calibration.py:52]:
```suggestion
        self.threshold_min_max = 2000 #TODO 
```

File: parallax/probe_calibration.py [hunk 2]
```diff
@@ -3,125 +3,296 @@
...
+        
+        Returns:
+            bool: True if there are enough points, False otherwise.
+        """ 
+        # End Criteria: 
+        # 1. distance maxX-minX, maxY-minY, maxZ-minZ
+        # 2. transM_LR difference in some epsilon value
+        # 3. L2 error (Global and Exp) is less than some values (e.g. 20 mincrons)
+        if self._is_criteria_met_points_min_max():
+            logger.debug("_is_criteria_met_points_min_max True")
```
Comment (jsiegle) [inline parallax/probe_calibration.py:244]: Should there be a check so this only gets printed once?

File: parallax/probe_calibration.py [hunk 3]
```diff
@@ -3,125 +3,296 @@
...
+        # update points in the file
+        self.stage = stage
+        self._update_local_global_point()
+        # get whole list of local and global points in pd format
+        local_points, global_points = self._get_local_global_points()
+        self.model_LR, self.transM_LR = self._get_transM_LR(local_points, global_points)
+        self.LR_err_L2_current = self._l2_error_current_point() 
+        self._update_min_max_x_y_z()  # update min max x,y,z 
 
+        # update transformation matrix and averall LR in UI
```
Comment (jsiegle) [inline parallax/probe_calibration.py:276]:
```suggestion
        # update transformation matrix and overall LR in UI
```

File: parallax/probe_calibration.py (hunk 1 above)
Comment (hannalee2) [reply parallax/probe_calibration.py]: Fixed! I guess I have a dyslexia! It took few minutes to see what is difference.

File: parallax/probe_calibration.py (hunk 2 above)
Comment (hannalee2) [reply parallax/probe_calibration.py]: Fixed. Debug messages can be seen on the log file only by the programmer if it is set to 'logger.setLevel(logging.DEBUG)'. User cannot see those message from the terminal prompt. No need to worry too much for the debug msg. I changed the debug message anyway! 
===============================
        if self._is_criteria_met_points_min_max():
            if self._is_criteria_met_transM():
                if self._is_criteria_met_l2_error():
                    logger.debug("Enough points gathered.")
                    return True
===============================

File: parallax/probe_calibration.py (hunk 3 above)
Comment (hannalee2) [reply parallax/probe_calibration.py]: Fixed

**Comment by jsiegle** [comment]
All looks good!

**Pull request merged**